frontend/out/production/frontend/frontend

```diff
@@ -0,0 +1,308 @@
+        request.addRequestDataEntry(ParameterType.DATEOFRETURN, dateOfReturn);
+
+        try {
+            sendRequestToSequencer(OperationCode.RETURN_ITEM, request);
+        } catch (Exception ex) {
+            return ex.getMessage();
+        }
+        return getResults();
+    }
+    public String createFindItem(String customerID, String itemName) {
+        String userPrefix = customerID.substring(0, 2);
+        ClientRequest request = null;
+        if(userPrefix.equalsIgnoreCase("qc")) {
+            request = new ClientRequest(OperationCode.FIND_ITEM, Location.QUEBEC, UserType.MANAGER);
+        }
+        if(userPrefix.equalsIgnoreCase("bc")){
+            request = new ClientRequest(OperationCode.FIND_ITEM, Location.BRITISHCOLUMBIA, UserType.MANAGER);
+        }
+        if(userPrefix.equalsIgnoreCase("on")){
+            request = new ClientRequest(OperationCode.FIND_ITEM, Location.ONTARIO, UserType.MANAGER);
+        }
+
+        request.addRequestDataEntry(ParameterType.CLIENTID, customerID);
+        request.addRequestDataEntry(ParameterType.ITEMNAME, itemName);
+
+        try {
+            sendRequestToSequencer(OperationCode.FIND_ITEM, request);
+        } catch (Exception ex) {
+            return ex.getMessage();
+        }
+        return getResults();
+    }
+
+    public String createExchangeItem(String customerID, String newItemID, String oldItemID, String dateOfExchange) {
+        String userPrefix = customerID.substring(0, 2);
+        ClientRequest request = null;
+        if(userPrefix.equalsIgnoreCase("qc")) {
+            request = new ClientRequest(OperationCode.EXCHANGE_ITEM, Location.QUEBEC, UserType.MANAGER);
+        }
+        if(userPrefix.equalsIgnoreCase("bc")){
+            request = new ClientRequest(OperationCode.EXCHANGE_ITEM, Location.BRITISHCOLUMBIA, UserType.MANAGER);
+        }
+        if(userPrefix.equalsIgnoreCase("on")){
+            request = new ClientRequest(OperationCode.EXCHANGE_ITEM, Location.ONTARIO, UserType.MANAGER);
+        }
+
+        request.addRequestDataEntry(ParameterType.CLIENTID, customerID);
+        request.addRequestDataEntry(ParameterType.NEWITEMID, newItemID);
+        request.addRequestDataEntry(ParameterType.OLDITEMID, oldItemID);
+        request.addRequestDataEntry(ParameterType.DATEOFEXCHANGE, dateOfExchange);
+
+        try {
+            sendRequestToSequencer(OperationCode.EXCHANGE_ITEM, request);
+        } catch (Exception ex) {
+            return ex.getMessage();
+        }
+        return getResults();
+    }
+
+
+    @Override
+    public void softwareFailure(String managerID) {
+        requiredAnswersForAgreement--;
+    }
+
+    @Override
+    public void replicaCrash(String managerID) {
+        requiredAnswersForAgreement--;
+    }
+
+
+    private void sendRequestToSequencer(OperationCode method, ClientRequest clientRequest) throws Exception {
+        ByteArrayOutputStream bStream = new ByteArrayOutputStream();
```

Comment:
This entire method eeds to be refactored to use the socketWrapper. Also we are only using ClientRequest and ReplicaResponses for communication. Message requests are ONLY for FE-RM communication and RM-Replica communication